channeldb/channel.go

```diff
@@ -1879,12 +1879,18 @@ func (c *OpenChannel) UpdateCommitment(newCommitment *ChannelCommitment,
 			return err
 		}
 
-		// Get the bucket where settled htlcs are recorded.
-		finalHtlcsBucket, err := fetchFinalHtlcsBucketRw(
-			tx, c.ShortChannelID,
-		)
-		if err != nil {
-			return err
+		// Get the bucket where settled htlcs are recorded if the user
+		// opted in to storing this information.
+		var finalHtlcsBucket kvdb.RwBucket
+		if c.Db.parent.storeFinalHtlcResolutions {
```

Comment:
will there be an issue to read `c.Db.parent.storeFinalHtlcResolutions` inside `c.Lock()`?

Reply:
What kind of issue are you thinking of with this non-blocking read?